build_tools/upload_logs_to_s3.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+import os
+import glob
+import subprocess
+
+
+def run_aws_cp(src, dest, content_type=None):
+    cmd = (
+        ["aws", "s3", "cp", src, dest, "--recursive"]
+        if os.path.isdir(src)
+        else ["aws", "s3", "cp", src, dest]
+    )
+    if content_type:
+        cmd += ["--content-type", content_type]
+    try:
+        print(f"[INFO] Running: {' '.join(cmd)}")
+        subprocess.run(cmd, check=True)
+    except subprocess.CalledProcessError as e:
+        print(f"[ERROR] Failed to upload {src} to {dest}: {e}")
+
+
+def upload_logs_to_s3(bucket_name, run_id, amdgpu_family, log_dir="build/logs"):
+    s3_base_path = f"s3://{bucket_name}/{run_id}/logs/{amdgpu_family}"
+
+    if not os.path.isdir(log_dir):
+        print("[INFO] No logs directory found. Creating it.")
+        os.makedirs(log_dir)
+        return
```

Comment:
Form the previous discussion, I don't see a case why you want to create an empty log dir. If the directory isn't existing there will be nothing in it and thus there is nothing to upload.